### PR TITLE
WIP: Adding a simple docker compose file for testing the proxy

### PR DESCRIPTION
**Pull request by arschles**
Still TODO:

- [ ] Document how to use this docker compose file to run tests

### CODE REVIEW

**Comment by marwan-at-work** [comment]
ops, accidentally hit update on this one. Thought it was my pr :)

**Comment by arschles** [comment]
>ops, accidentally hit update on this one. Thought it was my pr :)

@marwan-at-work no worries! I hit update frequently on other peoples' PRs

File: cmd/proxy/Dockerfile [hunk 1]
```diff
@@ -15,13 +15,19 @@ WORKDIR $GOPATH/src/github.com/gomods/athens
 ADD . .
 RUN cd cmd/proxy && buffalo build -s -o /bin/app
 
-FROM alpine
-RUN apk add --no-cache bash
-RUN apk add --no-cache ca-certificates
+RUN curl -o go1.11beta2.linux-amd64.tar.gz https://dl.google.com/go/go1.11beta2.linux-amd64.tar.gz
```
Comment (arschles) [inline cmd/proxy/Dockerfile]: Note to self: I'll need to update this to `go1.11beta3` now that #404 is merged

File: contrib/proxy/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on:
+      - redis
+    ports:
+      - "3000:3000"
+    environment:
+      - "ATHENS_STORAGE_TYPE=disk"
+      - "ATHENS_DISK_STORAGE_ROOT=/usr"
+    networks:
+      - main
+  redis:
```
Comment (arschles) [inline contrib/proxy/docker-compose.yml]: Note to self: after #400 is merged, redis won't be needed unless it's configured

**Comment by arschles** [comment]
This should wait until #437 is in

**Comment by manugupt1** [comment]
https://github.com/gomods/athens/pull/437 has been merged in @arschles

File: contrib/proxy/docker-compose.yml (hunk 2 above)
Comment (robjloranger) [reply contrib/proxy/docker-compose.yml]: Merged

**Comment by robjloranger** [comment]
note: #536 added working docker file for proxy

File: contrib/proxy/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+version: '3'
+services:
+  proxy:
+    image: gopackages/proxy
```
Comment (marwan-at-work) [inline contrib/proxy/docker-compose.yml]: how come our github's org is `gomods` but our docker's org is `gopackages`? 😄

Comment (marpio) [reply contrib/proxy/docker-compose.yml]: I don't think it is :-) https://hub.docker.com/r/gomods/

Comment (robjloranger) [reply contrib/proxy/docker-compose.yml]: that's the old one

Comment (arschles) [reply contrib/proxy/docker-compose.yml]: My bad - I'll fix this 👍

**Comment by arschles** [comment]
Ok, I've let this sit for a crazy long time and since #537, #437 and #400 have been merged, I'm gonna close this. Thanks so much for reviewing this and keeping up on is @robjloranger @marwan-at-work and @manugupt1! Apologies for not keeping up on it. You folks rock 😄

**Pull request closed without merging**